controllers/tlspolicy_controller.go

```diff
@@ -198,26 +202,44 @@ func (r *TLSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	gatewayEventMapper := mappers.NewGatewayEventMapper(mappers.WithLogger(r.Logger().WithName("gatewayEventMapper")), mappers.WithClient(mgr.GetClient()))
 
+	issuerStatusChangedPredicate := predicate.Funcs{
+		UpdateFunc: func(ev event.UpdateEvent) bool {
+			oldPolicy, ok := ev.ObjectOld.(certmanagerv1.GenericIssuer)
+			if !ok {
+				return false
+			}
+			newPolicy, ok := ev.ObjectNew.(certmanagerv1.GenericIssuer)
+			if !ok {
+				return false
+			}
+			oldStatus := oldPolicy.GetStatus()
+			newStatus := newPolicy.GetStatus()
+			return !reflect.DeepEqual(oldStatus, newStatus)
+		},
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.TLSPolicy{}).
+		Owns(&certmanagerv1.Certificate{}).
```

Comment:
if we start watching cert-manager API's, we need to make sure the CRD exists. 

check implementation of `kuadrantgatewayapi.IsCertManagerInstalled(r.RestMapper)` from https://github.com/Kuadrant/kuadrant-operator/pull/727 and we may want to add this before adding the watch. 

thanks @mikenairn for pointing this out.

Reply:
Sure, I can update to include this once https://github.com/Kuadrant/kuadrant-operator/pull/727 merges

Reply:
Actually I've added a similar commit for now to not wait for https://github.com/Kuadrant/kuadrant-operator/pull/727. If it merges before this one, I'll do a rebase instead 👍